src/mam4xx/wet_dep.hpp

```diff
@@ -1335,48 +1330,228 @@ void WetDeposition::compute_tendencies(
         totcond[k] = q_liq[k] + q_ice[k];
       });
   team.team_barrier();
-  wetdep::clddiag(nlev, temperature.data(), pmid.data(), pdel.data(),
-                  cmfdqr.data(), evapc.data(), cldt.data(), cldcu.data(),
-                  cldst.data(), evapr.data(), prain.data(), cldv.data(),
-                  cldvcu.data(), cldvst.data(), rain.data());
+
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, 1), KOKKOS_CLASS_LAMBDA(int k) {
+        wetdep::clddiag(nlev, temperature.data(), pmid.data(), pdel.data(),
+                        cmfdqr.data(), evapc.data(), cldt.data(), cldcu.data(),
+                        cldst.data(), evapr.data(), prain.data(), cldv.data(),
+                        cldvcu.data(), cldvst.data(), rain.data());
+      });
+  team.team_barrier();
   ColumnView dlf = diags.total_convective_detrainment;
+
+  // I don't have a good way to integrate the values computed for each
+  // level half way through the next Kokkos::parallel_for as the function
+  // calc_sfc_flux does.
+  // TODO: Determine a better way to implement calc_sfc_flux without
+  // team_fence() to avoid locking if the team size is not nlev;
+  EKAT_KERNEL_REQUIRE(team.team_size() == nlev);
```

Comment:
Let me know if you want a sounding wall or a chat about this.

Reply:
Have you looked into Kokkos's own [parallel_scan](https://kokkos.github.io/kokkos-core-wiki/API/core/parallel-dispatch/parallel_scan.html)?

Reply:
I think that since we are already in a team parallel loop, then Kokkos paralel_scan can not be used as we have used up the two levels of parallelism Kokkos supports.  But I could be wrong about that.   Given what we are learning about team parallel loops, maybe having the loop count match the team size would allow the parallel_scan to work. But that might be relying on unsupported behavior.

Reply:
Yeah, I wouldn't think that a parallel_scan inside a parallel_for is a good idea. It sounds to me like the right answer is (eventually) to restructure this code to be more sensible in a parallel environment. But if we can get this working by stipulating that the number of threads per team is `nlev`, I think that's a reasonable short-term solution.

Reply:
That may be fine. I'm not sure what role `convproc` is going to play in the final product. I suspect we'll be mining it piecemeal, in which case we'll likely be restructuring it anyway.